lib/util/bbcode.php

```diff
@@ -172,6 +172,12 @@ function linkifyYouTubeURLs($text)
     // https://www.youtube.com/watch?v=1YiNYWpwn7o
     // www.youtube.com/watch?v=Yjba9rvs4iU
 
+    // Don't linkify this text if it's a shortcode construct.
+    $linkedPattern = '/\[.*=.*\].*\[\/.*\]/';
+    if (preg_match($linkedPattern, $text) == 1) {
+        return $text;
+    }
```

Comment:
I haven't given this a run yet but from how it reads this would omit parsing _any_ youtube urls if there is at least one (url) shortcode construct with an attribute in the whole text?
I assumed that the pattern below has to be extended so that may not be preceded by indication of an enclosing shortcode.
The same should be applied to the other embed media parsers (twitch, imgur) as those would cause a similar problem.

Reply:
I'm not sure what you mean. This code exists here because there's no reason to linkify text that exists in a tag. The function seems to me to only be for wild links, so it stands to reason that it should return early if the input is not a wild link.

Are you suggesting we apply the same fix to the other linkify functions, or are you suggesting I modify the original monster regex? Doing the latter is, in my opinion, unnecessary and rather dangerous (after all, [we know the old adage about those who use regular expressions to solve problems](https://blog.codinghorror.com/regular-expressions-now-you-have-two-problems/)).

Reply:
Gave this another run to see what I meant there - can be confirmed when opening `/test/bbcode.php` in the browserfor smoke testing. It will not render any YouTube embeds anymore when a shortcode pattern can be found in the whole text that is to be parsed.

It doesn't do that line by line or for each match (globally, greedy).

I'll add the example from https://github.com/RetroAchievements/RAWeb/issues/818 to this PR and see if i can make it work with the existing monster regex.

I agree that it's dangerous to fiddle with regexes but we shouldn't have regex to parse bbcode/shortcode in the first place. Just trying to deal with what we already have :) Going by the simple test page should give at list _some_ confidence that nothing else breaks.

Reply:
Turns out we already have a negative lookahead in that regex which works when it finds a youtube url within an `<a>` tag.
All it's missing is a negative lookahead for shortcodes, too.

https://github.com/RetroAchievements/RAWeb/blob/master/lib/util/bbcode.php#L187-L193

I'll fix that for all the embed options.